pilot/pkg/networking/plugin/authz/authorization.go

```diff
@@ -94,50 +94,31 @@ func (p Plugin) buildFilter(in *plugin.InputParams, mutable *networking.MutableO
 		return
 	}
 
-	switch in.ListenerProtocol {
-	case networking.ListenerProtocolTCP:
-		option.Logger.AppendDebugf("building filters for TCP listener protocol")
-		tcpFilters := b.BuildTCP()
-		if in.Node.Type == model.Router {
-			// For gateways, due to TLS termination, a listener marked as TCP could very well
-			// be using a HTTP connection manager. So check the filterChain.listenerProtocol
-			// to decide the type of filter to attach
-			httpFilters := b.BuildHTTP()
-			for cnum := range mutable.FilterChains {
-				if mutable.FilterChains[cnum].ListenerProtocol == networking.ListenerProtocolHTTP {
-					option.Logger.AppendDebugf("added %d HTTP filters to gateway filter chain %d", len(httpFilters), cnum)
-					mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, httpFilters...)
-				} else {
-					option.Logger.AppendDebugf("added %d TCP filters to gateway filter chain %d", len(tcpFilters), cnum)
-					mutable.FilterChains[cnum].TCP = append(mutable.FilterChains[cnum].TCP, tcpFilters...)
-				}
+	// We will lazily build filters for tcp/http as needed
```

Comment:
do we also no longer need the `ListenerProtocolAuto`?

Reply:
ListenerProtocol Auto = FilterChainProtocols TCP and HTTP.

Now the plugins only act on filter chain protocol instead of listener protocol, so we only handle need to TCP and HTTP

Reply:
is ListenerProtocolAuto is needed on outbound listener only, for http sniffing? and here the PR is changing inbound, that's why we don't need to handle?